contain-intrinsic-* missing properties (#21110)
mdn/content: files/en-us/web/css/contain-intrinsic-inline-size/index.md

---
title: contain-intrinsic-inline-size
slug: Web/CSS/contain-intrinsic-inline-size
tags:
  - CSS
  - CSS Logical Property
  - CSS Property
  - Reference
  - experimental
  - contain-intrinsic-inline-size
page-type: css-property
browser-compat: css.properties.contain-intrinsic-inline-size
---

{{CSSRef}} {{SeeCompatTable}}

> The **`contain-intrinsic-inline-size`** [CSS](/en-US/docs/Web/CSS) [logical property](/en-US/docs/Web/CSS/CSS_Logical_Properties) defines the inline-size of an element that will be used for layout when it is subject to [size containment](/en-US/docs/Web/CSS/CSS_Containment#size_containment).

Reviewer: ```suggestion
The **`contain-intrinsic-inline-size`** [CSS](/en-US/docs/Web/CSS) [logical property](/en-US/docs/Web/CSS/CSS_Logical_Properties) defines the inline-size of an element that a browser can use for layout when the element is subject to [size containment](/en-US/docs/Web/CSS/CSS_Containment#size_containment).
```

> 

Reviewer: All suggestions in the `contain-intrinsic-block-size` page can be applied to the text below

Author: Thanks. I think I have captured them.

> The inline-size is the size of the element in the dimension parallel to the flow of text within a line.
> In a horizontal [writing mode](/en-US/docs/Web/CSS/writing-mode) like standard English this is the horizontal dimension (width); for a vertical writing mode it is the vertical dimension.

Author: ```suggestion
Inline-size is the size of the element in the dimension parallel to the flow of text within a line.
In a horizontal [writing mode](/en-US/docs/Web/CSS/writing-mode) like standard English, inline size is the horizontal dimension (width); for a vertical writing mode, inline size is the vertical dimension.
```

## Syntax

```css
/* Keyword values */
contain-intrinsic-inline-size: none;

/* <length> values */
contain-intrinsic-inline-size: 1000px;
contain-intrinsic-inline-size: 10rem;

/* auto <length> */
contain-intrinsic-inline-size: auto 300px;

/* Global values */
contain-intrinsic-inline-size: inherit;
contain-intrinsic-inline-size: initial;
contain-intrinsic-inline-size: revert;
contain-intrinsic-inline-size: unset;
```

### Values

> The following values may be specified for an element.

Author: ```suggestion
The following values can be specified for the intrinsic inline size of an element:
```

- `none`
  - : The element has no intrinsic inline-size.
- `<length>`
  - : The element has the specified inline-size ({{cssxref("&lt;length&gt;")}}).

> - `auto <length>`
>   - : A remembered value of the "normally rendered" element inline-size if one exists and the element is skipping its contents (for example, when it is offscreen); otherwise the specified `<length>`.

Author: ```suggestion
- `auto <length>`
  - : When the element is in size containment and skipping its contents (for example, when it is offscreen and `content-visibility: auto` is set) the inline size is remembered from the actual size of the element when it was last able to render its child elements.
    If the element has never rendered its child elements and hence has no remembered value for the normally rendered element size, or if it is not skipping its contents, the inline size is the specified `<length>`.
```

## Description

The property is commonly applied alongside elements that can trigger size containment, such as [`contain: size`](/en-US/docs/Web/CSS/contain) and [`content-visibility`](/en-US/docs/Web/CSS/content-visibility).

> Size containment allows a user agent to layout an element as though it had a fixed size, preventing unnecessary reflows by avoiding the re-rendering of child elements to determine the actual size (thereby improving user experience).

Reviewer 2: ```suggestion
Size containment allows a user agent to lay out an element as though it had a fixed size, preventing unnecessary reflows by avoiding the re-rendering of child elements to determine the actual size (thereby improving user experience).
```

tags:
  - CSS
  - CSS Logical Property
  - CSS Property
  - Reference
  - experimental
  - contain-intrinsic-inline-size
page-type: css-property
browser-compat: css.properties.contain-intrinsic-inline-size
---

{{CSSRef}} {{SeeCompatTable}}

The **`contain-intrinsic-inline-size`** [CSS](/en-US/docs/Web/CSS) [logical property](/en-US/docs/Web/CSS/CSS_Logical_Properties) defines the inline-size of an element that a browser can use for layout when the element is subject to [size containment](/en-US/docs/Web/CSS/CSS_Containment#size_containment).

Inline-size is the size of the element in the dimension parallel to the flow of text within a line.
In a horizontal [writing mode](/en-US/docs/Web/CSS/writing-mode) like standard English, inline size is the horizontal dimension (width); for a vertical writing mode, inline size is the vertical dimension.

## Syntax

```css
/* Keyword values */
contain-intrinsic-inline-size: none;

/* <length> values */
contain-intrinsic-inline-size: 1000px;
contain-intrinsic-inline-size: 10rem;

/* auto <length> */
contain-intrinsic-inline-size: auto 300px;

/* Global values */
contain-intrinsic-inline-size: inherit;
contain-intrinsic-inline-size: initial;

> contain-intrinsic-inline-size: revert;
> contain-intrinsic-inline-size: unset;

Author: ```suggestion
contain-intrinsic-inline-size: revert;
contain-intrinsic-inline-size: revert-layer;
contain-intrinsic-inline-size: unset;
```
